sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -244,11 +244,19 @@ statement
     | SET TIME ZONE interval                                           #setTimeZone
     | SET TIME ZONE timezone=(STRING | LOCAL)                          #setTimeZone
     | SET TIME ZONE .*?                                                #setTimeZone
+    | SET configKey (EQ value=.+)?                                     #setConfiguration
     | SET .*?                                                          #setConfiguration
+    | RESET configKey?                                                 #resetConfiguration
     | RESET .*?                                                        #resetConfiguration
     | unsupportedHiveNativeCommands .*?                                #failNativeCommand
     ;
 
+configKey
+    : IDENTIFIER (('.' | ':') IDENTIFIER)*
```

Comment:
Currently, this rule leads to some test failures;
```
sbt.ForkMain$ForkError: org.apache.spark.sql.catalyst.parser.ParseException: 
Expected format is 'SET', 'SET key', or 'SET key=value'. If you want to include special characters in key and value, please use quotes or string literal, e.g., SET "ke y"=value.(line 1, pos 0)

== SQL ==
set hive.fetch.task.conversion=more
^^^
```
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/126674/testReport/org.apache.spark.sql.hive.execution/HiveCompatibilitySuite/binary_constant/

This is because `IDENTIFIER` seems to be conflict with the existing ANTLR token, e.g., `fetch` in the example above. I'm currently looking for another smart rule to avoid it. Please let me know if someone has better idea..

Reply:
maybe we should replace `IDENTIFIER (('.' | ':') IDENTIFIER)*` with
```
( ~' ' | ~'=' )+
```

Reply:
or replace `IDENTIFIER` with `IDENTIFIER | nonReserved | strictNonReserved`

`nonReserved | strictNonReserved` matches all the keywords